common/network-common/src/test/java/org/apache/spark/network/util/TransportFrameDecoderSuite.java

```diff
@@ -47,6 +51,88 @@ public void testFrameDecoding() throws Exception {
     verifyAndCloseDecoder(decoder, ctx, data);
   }
 
+  @Test
+  public void testConsolidationForDecodingNonFullyWrittenByteBuf() {
```

Comment:
If I understand correctly, this is testing that consolidation is reducing the amount of memory needed to hold a frame? But since you're writing just 1 MB to the decoder, that's not triggering consolidation, is it? 

Playing with `CompositeByteBuf`, it adjusts the internal capacity based on the readable bytes of the components, but the component buffers remain unchanged, so still holding on to the original amount of memory:

```
scala> cb.numComponents()
res4: Int = 2

scala> cb.capacity()
res5: Int = 8

scala> cb.component(0).capacity()
res6: Int = 1048576
```

So I'm not sure this test is testing anything useful.

Also it would be nice not to use so many magic numbers.

Reply:
@vanzin I think the test should be refined. but I was quesion about your test.
CompositeByteBuf.capacity returns the last component endOffset, I think use the capacity for testing is ok.
https://github.com/netty/netty/blob/8fecbab2c56d3f49d0353d58ee1681f3e6d3feca/buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java#L730

Reply:
Maybe my question wasn't clear. I'm asking what part of Spark code is this test testing.

As far as I can see, it's testing netty code, and these are not netty unit tests.

Reply:
@vanzin it think this test is a little duplicate of `testConsolidationPerf`, we can just remove it. I will update soon. Sorry for that.